### ISSUE TITLE
Optimal parameters to apply on small chunks of data for streaming application

### ISSUE THREAD

**Issue by H-G-11** [timsainb/noisereduce#89]
Hi,

I am building an application that streams data from an audio input.
I am applying the noisereduce algorithm (torch version) to every chunk of 500ms audio data, but will probably go down to 20ms at some point.

It seems to me that this algorithm works great for one big audio file, but applying it to many small audio files each after another leads to the final filtered output to be of poor quality (the noise is not removed in the same way everywhere).

I am sure I can improve things by tunning the hyper-parameters. Would someone be so kind as telling me which ones should be optimized?
I am quite new to audio data, so I am not sure how I can tackle this issue.

Thank you so much for your answers and this great repository.

**Comment by timsainb** [comment]
Hi Hugues,

I assume you are using the stationary version of the algorithm? The stationary version of the algorithm should be the same for long vs short clips if you are providing the same noise clip as input. It doesn't make sense to perform non-stationary noise reduction because you are basically providing stationary input if the timescale is too short. 

If you have some metric of quality it is possible to search parameter space that way - e.g. training a prediction model on the output and seeing what set of parameters perform best. 

All the parameters are in the main readme. I would focus on the prop_decrease, time_constant_s, freq_mask_smooth_hz, time_mask_smooth_ms, sigmoid_slope_nonstationary, n_std_thresh_stationary

These all relate to how the mask is built. 

Best,
Tim

**Comment by DamienDeepgram** [comment]
Interested in how to get this working for streaming audio also, did you ever get something working @HuguesGallier ?
